src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -410,16 +418,229 @@ def on_request(self, request, **kwargs):
+                       management_group_id, deployment_name, deployment_properties, deployment_location)
+
+
+def deploy_arm_template_at_tenant_scope(cmd,
+                                        template_file=None, template_uri=None, parameters=None,
+                                        deployment_name=None, deployment_location=None,
+                                        no_wait=False, handle_extended_json_format=False):
+    return _deploy_arm_template_at_tenant_scope(cli_ctx=cmd.cli_ctx,
+                                                template_file=template_file, template_uri=template_uri, parameters=parameters,
+                                                deployment_name=deployment_name, deployment_location=deployment_location,
+                                                validate_only=False,
+                                                no_wait=no_wait, handle_extended_json_format=handle_extended_json_format)
+
+
+def validate_arm_template_at_tenant_scope(cmd,
+                                          template_file=None, template_uri=None, parameters=None,
+                                          deployment_name=None, deployment_location=None,
+                                          no_wait=False, handle_extended_json_format=False):
+    return _deploy_arm_template_at_tenant_scope(cli_ctx=cmd.cli_ctx,
+                                                template_file=template_file, template_uri=template_uri, parameters=parameters,
+                                                deployment_name=deployment_name, deployment_location=deployment_location,
+                                                validate_only=True,
+                                                no_wait=no_wait, handle_extended_json_format=handle_extended_json_format)
+
+
+def _deploy_arm_template_at_tenant_scope(cli_ctx,
+                                         template_file=None, template_uri=None, parameters=None,
+                                         deployment_name=None, deployment_location=None,
+                                         validate_only=False,
+                                         no_wait=False, handle_extended_json_format=False):
+    deployment_properties = None
+    if handle_extended_json_format:
+        deployment_properties = _prepare_deployment_properties_unmodified(cli_ctx=cli_ctx, template_file=template_file,
+                                                                          template_uri=template_uri,
+                                                                          parameters=parameters, mode='Incremental')
+    else:
+        deployment_properties = _prepare_deployment_properties(cli_ctx=cli_ctx, template_file=template_file,
+                                                               template_uri=template_uri,
+                                                               parameters=parameters, mode='Incremental')
```

Comment:
I see lot's of if handle_extended_json_format check in this file, may be you can add another method to wrap this logic

Reply:
I tried.. but didn't work out a cleaner solution.